### PR TITLE
Issue #199: Put the closure in heap to avoid the stack overflowed in `Debug`

### PR DESCRIPTION
**Pull request by streetycat**

### CODE REVIEW

File: src/component/cyfs-base/src/codec/raw/raw_types.rs [hunk 1]
```diff
@@ -757,7 +757,12 @@ impl<T: RawEncode> RawEncode for HashSet<T> {
     ) -> BuckyResult<&'a mut [u8]> {
         let ulen = USize(self.len());
         let mut buf = ulen.raw_encode(buf, purpose)?;
-        for e in self {
+
+        // stable sort
+        let mut values: Vec<&T> = self.iter().collect();
```
Comment (lurenpluto) [inline src/component/cyfs-base/src/codec/raw/raw_types.rs:762]: This should be the change mentioned by https://github.com/buckyos/CYFS/issues/256, it should not be in this pull request

File: src/meta/cyfs-meta/src/creator.rs
Comment (lurenpluto) [inline src/meta/cyfs-meta/src/creator.rs:1]: This file only seems to have changed in format, and I compared it to see no substantive changes, and the changes seem to have nothing to do with the problem the target is trying to fix

**Review by lurenpluto — COMMENTED** [review]
What are the key changes to this fix? It looks like it's all unrelated changes

File: src/component/cyfs-base/src/codec/raw/raw_types.rs (hunk 1 above)
Comment (streetycat) [reply src/component/cyfs-base/src/codec/raw/raw_types.rs:762]: Sort the `values` in ascending order, and encode the elements into the buffer. for Issue #256

Comment (lurenpluto) [reply src/component/cyfs-base/src/codec/raw/raw_types.rs:762]: I mean it should be included in a new request for issue #256, and in the entire pull request, it seems, this is the only substantive change?

File: src/meta/cyfs-meta/src/creator.rs
Comment (streetycat) [reply src/meta/cyfs-meta/src/creator.rs:1]: Yes, It's only format change. It occurs when I merged the main branch to my local branch, and the IDE changed the format automatically, I think we shoud define the standuard configure of format for IDE.

**Pull request closed without merging**